server/routes_test.go

```diff
@@ -477,3 +477,46 @@ func TestRouteUseIPv6(t *testing.T) {
 		t.Fatal("Server failed to start route accept loop")
 	}
 }
+
+func TestClientConnectToRoutePort(t *testing.T) {
+	opts := DefaultOptions
+	opts.Cluster.NoAdvertise = true
+	s := RunServer(&opts)
+	defer s.Shutdown()
+
+	url := fmt.Sprintf("nats://%s:%d", opts.Cluster.Host, opts.Cluster.Port)
+	clientURL := fmt.Sprintf("nats://%s:%d", opts.Host, opts.Port)
+	total := 100
+	for i := 0; i < total; i++ {
+		nc, err := nats.Connect(url)
+		if err == nil {
+			// It is possible that the client reconnects because
+			// it gets from the initial connect to the route the
+			// connectedUrl array and may be able to reconnects
+			// to the client URL.
+			// If connected, it should be to the client URL
+			if nc.ConnectedUrl() != clientURL {
```

Comment:
This test is based on current behavior of NATS Go client, which I think may have an issue. On initial connect, the client gets possibly a list of connect URLs suitable for clients. It adds it to the pool and shuffle the pool (unless NoRandomize option is set). I think this is wrong to do that in the initial connect loop. Need to discuss that with the group and see what other clients do.
If we were not to shuffle the pool on processing of initial INFO when getting new URLs from the INFO protocol, this server test would only have one case which is that connect should always succeed.